lua/autorun/server/sv_restricted_propcore_functions.lua

```diff
@@ -0,0 +1,52 @@
+-- Restricted Propcore Functions
+---------------------------------------------------------------------------------------
+local RESTRICTED_FUNCTIONS = {
+  'propSpawn(sn)',
+  'propSpawn(en)',
+  'propSpawn(svn)',
+  'propSpawn(evn)',
+  'propSpawn(san)',
+  'propSpawn(ean)',
+  'propSpawn(svan)',
+  'propSpawn(evan)',
+  'seatSpawn(sn)',
+  'seatSpawn(svan)',
+  'setPos(e:v)',
+  'propManipulate(e:vannn)',
+  'reposition(e:v)'
+}
+---------------------------------------------------------------------------------------
+
+-- Rank Restriction Tables
+---------------------------------------------------------------------------------------
+-- Admin rank restricted functions
+local ADMIN_RESTRICTED_FUNCTIONS = {}
+
+-- Regular rank restricted functions
+local REGULAR_RESTRICTED_FUNCTIONS = ADMIN_RESTRICTED_FUNCTIONS
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(sn)']           = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(en)']           = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(svn)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(evn)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(san)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(ean)']          = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(svan)']         = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propSpawn(evan)']         = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['seatSpawn(sn)']           = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['seatSpawn(svan)']         = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['setPos(e:v)']             = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['propManipulate(e:vannn)'] = {build = true, pvp = true}
+REGULAR_RESTRICTED_FUNCTIONS['reposition(e:v)']         = {build = true, pvp = true}
+--REGULAR_RESTRICTED_FUNCTIONS['propBreak(e:)']         = {build = true, pvp = true}
+
+-- User rank restricted functions
+local USER_RESTRICTED_FUNCTIONS = REGULAR_RESTRICTED_FUNCTIONS
```

Comment:
@brandonsturgeon 

Adding a new rank would be as so:
```lua
local NEWRANK_RESTRICTED_FUNCTIONS = RANK_THAT_SUPERSEDES_THIS_RESTRICTED_FUNCTIONS
NEWRANK_RESTRICTED_FUNCTIONS[function] = {build = t/f, pvp = t/f}
...

local RESTRICTED_FUNTIONS_BY_RANK = {
    admin = ...
    regular = ...
    user = ...
    newRank = NEWRANK_RESTRICTED_FUNCTIONS
}
```
The good thing about this is that you could base a rank off of  the `user` rank, and then override/add any methods for specific allowances.

Reply:
Assuming that Regulars inherit from Devotees, they would initially have the same restriction table.
However, to allow something to them that isn't allowed to Devotees:

Where Devotees would have:
```lua
DEVOTEE_RESTRICTED_FUNCTIONS["function"] = {build = true, pvp = true}
```

The Regular setup might look like:
```
REGULAR_RESTRICTED_FUNCTIONS = DEVOTEE_RESTRICTED_FUNCTIONS

...

REGULAR_RESTRICTED_FUNCTIONS["function"] = {build = false, pvp = false}
```
The entry in their table would just be overwritten.

Reply:
Just found out these boys are passed by reference. We'll need a table copy function.

Reply:
Okay, that's done.